mmf/models/movie_mcan.py

```diff
@@ -0,0 +1,270 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import copy
+from typing import Any, Dict, List, Optional, Tuple
+
+import torch
+
+from mmf.common.registry import registry
+from mmf.common.typings import DictConfig
+from mmf.models.base_model import BaseModel
+from mmf.modules.embeddings import (
+    PreExtractedEmbedding,
+    TextEmbedding,
+    TwoBranchEmbedding,
+)
+from mmf.modules.encoders import ImageFeatureEncoder
+from mmf.modules.layers import BranchCombineLayer, ClassifierLayer
+from mmf.utils.general import filter_grads
+
+
+@registry.register_model("movie_mcan")
+class MoVieMcan(BaseModel):
```

Comment:
Something to keep in mind but not important as of now for speed:

Since the code has a lot of duplication w.r.t. to Pythia class, I would just take out the common parts and create a fusion mixin. And then use that mixin both in Pythia and here to reuse common parts. then, it also won't look like that MovieMCAN is built on top of Pythia.